modules/test/playwright/pages/portal-search-web/SearchPage.ts

```diff
@@ -119,43 +119,63 @@ export class SearchPage {
 		).toHaveAttribute('aria-current', 'page');
 	}
 
-	async selectSearchFacetCheckbox(searchFacetCheckbox: Locator) {
-		await searchFacetCheckbox.check();
+	async selectSearchFacetCheckbox(
+		searchFacetCheckbox: Locator,
+		value: boolean = true
+	) {
+		if (value) {
+			await searchFacetCheckbox.check();
+
+			await expect(searchFacetCheckbox).toBeChecked();
+		}
+		else {
+			await searchFacetCheckbox.uncheck();
 
-		await expect(searchFacetCheckbox).toBeChecked();
+			await expect(searchFacetCheckbox).not.toBeChecked();
+		}
 	}
 
-	async selectSearchFacetLink(searchFacetLink: Locator) {
+	async selectSearchFacetLink(
+		searchFacetLink: Locator,
+		value: boolean = true
+	) {
 		await searchFacetLink.click();
 
 		await expect(searchFacetLink).not.toBeDisabled();
 
-		await expect(searchFacetLink).toHaveClass(/facet-term-selected/);
+		if (value) {
+			await expect(searchFacetLink).toHaveClass(/facet-term-selected/);
+		}
+		else {
+			await expect(searchFacetLink).not.toHaveClass(
+				/facet-term-selected/
+			);
+		}
 	}
 
-	async selectSearchOptionConfiguration(option: string, value: boolean) {
+	async selectSearchOptionConfigurations(options: string[], value: boolean) {
 		await this.searchOptionsConfigurationLink.click();
 
-		const configurationCheckbox = this.modalIFrame.locator(
-			`xpath=//*[text()[contains(.,'${option}')]]//input`
-		);
+		for (const option of options) {
+			const configurationCheckbox = this.modalIFrame.locator(
+				`xpath=//*[text()[contains(.,'${option}')]]//input`
+			);
 
-		const checked = await configurationCheckbox.isChecked();
+			const checked = await configurationCheckbox.isChecked();
 
-		if (value) {
-			if (!checked) {
-				await configurationCheckbox.check();
+			if (value && !checked) {
+				await this.selectSearchFacetCheckbox(
+					configurationCheckbox,
+					true
+				);
 			}
 
-			await expect(configurationCheckbox).toBeChecked();
-		}
-
-		if (!value) {
-			if (checked) {
-				await configurationCheckbox.uncheck();
+			if (!value && checked) {
+				await this.selectSearchFacetCheckbox(
+					configurationCheckbox,
+					false
+				);
```

Comment:
We don't need to check for `checked`. When calling `.check()` on a checkbox already checked, nothing will happen.

This can be simplified to:

```
await this.selectSearchFacetCheckbox(
    configurationCheckbox,
    value
);
```

Reply:
Oh wow, thank you! That cleans it up a lot 💯